llvm/include/llvm/Analysis/MemoryLocation.h

```diff
@@ -189,14 +189,16 @@ class LocationSize {
   bool operator==(const LocationSize &Other) const {
     return Value == Other.Value;
   }
-
   bool operator==(const TypeSize &Other) const {
-    return hasValue() && getValue() == Other;
+    return (*this == LocationSize::precise(Other));
```

Comment:
This change isn't related to the conversion case.  I noticed when writing the other equality operand that this idiom seems unsound - unless I'm misreading the code it allows an imprecise upper bound to compare equal with a typesize.  I can split this off if anyone asks, but I don't have a test case which shows it being actually problematic in practice.